### PR TITLE
[Modules] Bicep module to create Front Door WAF Policy (#3718)

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Based on PR #3718 
- New module added to create WAF Policy for Front Door related to https://github.com/Azure/ResourceModules/issues/3714

cc: @rajeshkaremane 

| Pipeline |
| - |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=137&branchName=users%2Falsehr%2F228005_DeployFrontDoorWAFRules_rajeshkaremane) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/network/front-door-web-application-firewall-policy/main.bicep [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    principalIds: roleAssignment.principalIds
+    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    condition: contains(roleAssignment, 'condition') ? roleAssignment.condition : ''
+    delegatedManagedIdentityResourceId: contains(roleAssignment, 'delegatedManagedIdentityResourceId') ? roleAssignment.delegatedManagedIdentityResourceId : ''
+    resourceId: frontDoorWAFPolicy.id
+  }
+}]
+
+@description('The name of the application gateway WAF policy.')
```
Comment (ahmadabdalla) [inline modules/network/front-door-web-application-firewall-policy/main.bicep:131]:
```suggestion
@description('The name of the Front Door WAF policy.')
```

File: modules/network/front-door-web-application-firewall-policy/main.bicep [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+output name string = frontDoorWAFPolicy.name
+
+@description('The resource ID of the application gateway WAF policy.')
```
Comment (ahmadabdalla) [inline modules/network/front-door-web-application-firewall-policy/main.bicep:134]:
```suggestion
@description('The resource ID of the Front Door WAF policy.')
```

File: modules/network/front-door-web-application-firewall-policy/main.bicep [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+output resourceId string = frontDoorWAFPolicy.id
+
+@description('The resource group the application gateway WAF policy was deployed into.')
```
Comment (ahmadabdalla) [inline modules/network/front-door-web-application-firewall-policy/main.bicep:137]:
```suggestion
@description('The resource group the Front Door WAF policy was deployed into.')
```

File: modules/network/front-door-web-application-firewall-policy/main.bicep [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+@description('Optional. Location for all resources.')
+param location string = 'global'
+
+@allowed([
+  'Standard_AzureFrontDoor'
+  'Premium_AzureFrontDoor'
+])
+@description('Optional. The pricing tier of the WAF profile.')
+param sku string = 'Premium_AzureFrontDoor'
```
Comment (ahmadabdalla) [inline modules/network/front-door-web-application-firewall-policy/main.bicep:18]: Should this default to standard?

File: modules/network/front-door-web-application-firewall-policy/.test/common/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    roleAssignments: [
+      {
+        roleDefinitionIdOrName: 'Reader'
+        principalIds: [
+          nestedDependencies.outputs.managedIdentityPrincipalId
+        ]
+        principalType: 'ServicePrincipal'
+      }
+    ]
+  }
```
Comment (ahmadabdalla) [inline modules/network/front-door-web-application-firewall-policy/.test/common/main.test.bicep:127]: Can we include the lock and SKU properties as part of this test file?

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]

File: modules/network/front-door-web-application-firewall-policy/main.bicep (hunk 4 above)
Comment (AlexanderSehr) [reply modules/network/front-door-web-application-firewall-policy/main.bicep:18]: This was from the original contributor. No clue. There seem to be a few differences such as the lack of ManagedRule definitions: https://learn.microsoft.com/en-us/azure/frontdoor/standard-premium/tier-comparison . Can't say I have much of an opinion about this one without knowing the service better.

**Review by rajeshkaremane — APPROVED** [review]

File: modules/network/front-door-web-application-firewall-policy/main.bicep (hunk 4 above)
Comment (AlexanderSehr) [reply modules/network/front-door-web-application-firewall-policy/main.bicep]: Updated it now and will re-run the pipeline

Comment (rajeshkaremane) [reply modules/network/front-door-web-application-firewall-policy/main.bicep]: Yes it is fine to use standard as default. I have added conditional deployment for ManagedRule section

File: modules/network/front-door-web-application-firewall-policy/.test/common/main.test.bicep (hunk 5 above)
Comment (AlexanderSehr) [reply modules/network/front-door-web-application-firewall-policy/.test/common/main.test.bicep:129]: done

**Review by rajeshkaremane — APPROVED** [review]

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**